cruise.umple/src/Umple_Code.ump

```diff
@@ -1880,8 +1880,15 @@ class UmpleClass
     return wasSet;
   }
 
+
+  internal const Integer CounterLimit = 25; //Arbitary Limit
+  Integer propagateCounter = CounterLimit; 
```

Comment:
You are using an instance variable for the propagateCounter. A better solution would have been to simply add an argument. The initial call sets it to 25; recursive calls reduce by1.
You are using an instance variable for the propagateCounter. A better solution would have been to simply add an argument. The initial call sets it to 25; recursive calls reduce by 1.

Reply:
I had considered passing the counter as an argument, but while debugging noticed that the method calls: notifySubclassesAncestorImmutable, propagateImmutabilityToAllRelationships and setAncestorIsImmutable, occur in a loop. This means that the counter will reset itself and will never reach the base case.